pkg/shipper/shipper.go

```diff
@@ -339,7 +340,7 @@ func (s *Shipper) Sync(ctx context.Context) (uploaded int, err error) {
 		return uploaded, errors.Errorf("failed to sync %v blocks", uploadErrs)
 	}
 
-	if s.uploadCompacted {
+	if uploadCompacted {
 		s.metrics.uploadedCompacted.Set(1)
```

Comment:
Do we need to reset this metric if `uploadCompacted` suddenly became false?

Reply:
I am not sure how this metric is designed and used tbh. But I can set it to 0 if it is disabled.

Reply:
I'm not sure too but I think let's call `.Reset()` instead https://pkg.go.dev/github.com/prometheus/client_golang@v1.16.0/prometheus#MetricVec.Reset to remove all previously set metrics.

Reply:
`.Reset()` is only available in `MetricVec` but this is a `Gauge`, not `GaugeVec`.